polygeist/test/lib/Dialect/Polygeist/TestReachingDefinitionsAnalysis.cpp

```diff
@@ -83,8 +85,15 @@ struct TestReachingDefinitionAnalysisPass
       // Print the reaching definitions for each operand.
       for (auto [index, operand] : llvm::enumerate(op->getOperands())) {
         llvm::errs() << " operand #" << index << "\n";
-        auto mods = reachingDef->getModifiers(operand);
-        auto pMods = reachingDef->getPotentialModifiers(operand);
+
+        Value val = UnderlyingValueAnalysis::getMostUnderlyingValue(
+            operand, [&](Value val) {
+              return solver.lookupState<UnderlyingValueLattice>(val);
+            });
+        assert(val && "expected an underlying value");
+
+        auto mods = reachingDef->getModifiers(val);
+        auto pMods = reachingDef->getPotentialModifiers(val);
```

Comment:
Is it possible to keep calling `getModifiers` and `getPotentialModifiers` on the operand like before and have the logic of `getMostUnderlyingValue` in reaching def analysis?

Reply:
I agree, it would be nice to have that directly incorporated into the reaching definition analysis, so not all analyses building on top of that need to add this logic.

Reply:
yes - addressed

Reply:
Good idea! Thanks for feedback.